### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by jforissier**
Fixing the easy ones.

### CODE REVIEW

**Comment by jforissier** [comment]
@vchong how do I trigger a Coverity scan on my project? Do I just need to push to the `coverity_scan` branch? (I have registered jforissier/optee_os already, but it seems it has to go through a review process first?).

File: core/tee/tee_svc_cryp.c [hunk 1]
```diff
@@ -906,6 +896,30 @@ static const struct tee_cryp_obj_type_props *tee_svc_find_type_props(
...
+/* Set an attribute on an object */
+static void set_attribute(struct tee_obj *o,
+			  const struct tee_cryp_obj_type_props *props,
+			  uint32_t attr)
+{
+	int idx = tee_svc_cryp_obj_find_type_attr_idx(attr, props);
+
+	if (idx < 0)
+		return;
+	o->have_attrs |= 1 << idx;
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Please use `BIT(idx)` instead of `1 << idx`

File: core/tee/tee_svc_cryp.c [hunk 2]
```diff
@@ -906,6 +896,30 @@ static const struct tee_cryp_obj_type_props *tee_svc_find_type_props(
...
+/* Get an attribute on an object */
+static uint32_t get_attribute(const struct tee_obj *o,
+			      const struct tee_cryp_obj_type_props *props,
+			      uint32_t attr)
+{
+	int idx = tee_svc_cryp_obj_find_type_attr_idx(attr, props);
+
+	if (idx < 0)
+		return 0;
+	return o->have_attrs & (1 << idx);
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Please use `BIT(idx)` instead of `1 << idx`

**Comment by vchong** [comment]
@jforissier Ref: https://github.com/OP-TEE/optee_os/compare/coverity_scan...vchong:coverity_scan?expand=1

1. Configure https://travis-ci.org/jforissier/optee_os/settings to `Build pushes`.
2. Replace the `secure` token in `.travis.yml` with yours from https://scan.coverity.com/projects/jforissier-optee_os/builds/new?tab=travis_ci.
3. Change my handle in `.travis.yml` to yours as necessary.
4. Commit changes and push to your fork's `coverity_scan` branch and it should trigger a scan.
5. Yes, you might not be able to see the defects until they approve your project, but you should still be able to see the scan triggered in Travis' log and view the scan summary on the Coverity page.

**Comment by vchong** [comment]
Please also note that the official `coverity_scan` branch is about 4-5 commits behind official `master` so your `coverity_scan` branch will be ahead of the official one if you're based on a more recent `master`.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
@vchong thanks for the detailed instructions 👍

**Comment by vchong** [comment]
@jforissier np.. glad to help :)

File: core/tee/tee_obj.c [hunk 3]
```diff
@@ -70,10 +70,11 @@ void tee_obj_close(struct user_ta_ctx *utc, struct tee_obj *o)
 
 void tee_obj_close_all(struct user_ta_ctx *utc)
 {
-	struct tee_obj_head *objects = &utc->objects;
+	struct tee_obj *o;
+	struct tee_obj *next;
 
-	while (!TAILQ_EMPTY(objects))
-		tee_obj_close(utc, TAILQ_FIRST(objects));
```
Comment (lackan) [inline core/tee/tee_obj.c]: Hi, Jerome,

Will the origin code use the pointer after free? I think about this, and can't find that. Can you explain something to me?

Comment (jforissier) [inline core/tee/tee_obj.c]: Hi @lackan,
I think you're right, the original code is correct. I suppose Coverity is confused by the use of `tqh_first` and `tqe_next`. This seems to be a common issue with Coverity (https://mail-index.netbsd.org/coverity-updates/2014/02/19/msg000045.html).

@vchong can we tell Coverity to ignore this kind of construct? Or at least ignore the particular occurrences?
I'll revert the bogus fixes.

Comment (vchong) [inline core/tee/tee_obj.c]: @jforissier It should be possible to submit a model to ignore the false positives. I'll take a closer look and update the Coverity project with it as soon as possible. To ignore these particular occurrences, we can just go to the defects list, select the CIDs and choose Ignore for the Action.

**Pull request merged**